scripts/installer.sh

```diff
@@ -0,0 +1,251 @@
+			OS="amazon-linux"
+			VERSION="$VERSION_ID"
+			PACKAGETYPE="yum"
+			;;
+		opensuse-leap)
+			OS="opensuse"
+			VERSION="leap/$VERSION_ID"
+			PACKAGETYPE="zypper"
+			;;
+		opensuse-tumbleweed)
+			OS="opensuse"
+			VERSION="tumbleweed"
+			PACKAGETYPE="zypper"
+			;;
+ 		arch)
+			echo "Please install Tailscale from the community repo:"
+			echo
+			echo "pacman -S tailscale"
+			echo "systemctl enable --now tailscaled"
+			exit 2
+			;;
+		manjaro)
+			echo "Please install Tailscale from the community repo:"
+			echo
+			echo "pacman -S tailscale"
+			echo "systemctl enable --now tailscaled"
+			exit 2
+			;;
+		alpine)
+			echo "Please install Tailscale from the alpine testing repo"
+			echo
+			echo "apk add tailscale"
+			echo "rc-update add tailscale"
+			exit 2
+			;;
+		nixos)
+			echo "Please add Tailscale to your NixOS configuration directly"
+			echo
+			echo "services.tailscale.enable = true;"
+			exit 2
+			;;
+		void)
+			echo "Please install Tailscale from the void current repo"
+			echo
+			echo "xbps-install tailscale"
+			exit 2
+			;;
+		gentoo)
+			echo "Please install Tailscale from portage"
+			echo
+			echo "emerge net-vpn/tailscale"
+			exit 2
+			;;
+		freebsd)
+			echo "Please install Tailscale using pkg"
+			echo
+			echo "pkg add tailscale"
+			exit 2
+			;;
+		openbsd)
+			echo "Tailscale does not currently have packages for OpenBSD."
+			echo "Please install Tailscale from source: https://github.com/tailscale/tailscale"
+			exit 2
+			;;
+		*)
+			echo "Your OS ($ID) is not supported by this script."
+			echo "Please contact us at support@tailscale.com for help."
+			exit 2
+			;;
+		# TODO: wsl?
+	esac
+
+	# Step 2: having detected an OS we support, is it one of the
+	# versions we support?
+	OS_UNSUPPORTED=
+	case "$OS" in
+		ubuntu)
+			if [ "$VERSION" != "16.04" -a
+				 "$VERSION" != "18.04" -a
+				 "$VERSION" != "19.10" -a
+				 "$VERSION" != "20.04" -a
+				 "$VERSION" != "20.10" -a
+				 "$VERSION" != "21.04"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		debian)
+			if [ "$VERSION" != "stretch" -a
+				 "$VERSION" != "buster" -a
+				 "$VERSION" != "bullseye" -a
+				 "$VERSION" != "sid"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		raspbian)
+			if [ "$VERSION" != "buster"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		centos)
+			if [ "$VERSION" != "7" -a
+				 "$VERSION" != "8"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		rhel)
+			if [ "$VERSION" != "8"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		amazon-linux)
+			if [ "$VERSION" != "2"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		opensuse)
+			if [ "$VERSION" != "leap/15.1" -a
+				 "$VERSION" != "leap/15.2" -a
+				 "$VERSION" != "tumbleweed"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		*)
+			OS_UNSUPPORTED=1
+	esac
+	if [ "$OS_UNSUPPORTED" = "1" ]; then
+		echo "$OS $VERSION is not a supported version of $OS for Tailscale."
+		echo "Please contact us at support@tailscale.com for help."
```

Comment:
This would be another great place to prompt people to check out [our supported Linux distros](https://tailscale.com/kb/1031/install-linux/) and suggest [static binaries](https://tailscale.com/kb/1053/install-static/) if it's not listed.

I think we only want folks to contact us if they think something is wrong/broken.